### ISSUE TITLE
[Cache] Fatal error:  Interface 'Psr\SimpleCache\CacheInterface' not found

### ISSUE THREAD

**Issue by AndrzejDunowski** [symfony/symfony#31720]
After upgrade from v4.2.9 to v4.3.0

```
Fatal error:  Interface 'Psr\SimpleCache\CacheInterface' not found in /var/www/clients/client1/web107/private/vendor/symfony/cache/Simple/AbstractCache.php on line 28
```

Helps

> composer require psr/simple-cache

**Comment by nicolas-grekas** [comment]
Correct: we removed the dependency from symfony/cache so you need to declare it explicitly in your app now. Maybe worth a CHANGELOG/UPGRADE entry? Would you mind sending a PR doing so?

**Comment by derrabus** [comment]
What troubles me a bit is that Symfony 4.2 applications that don't use the PSR interface directly suddenly fail after the upgrade, for instance when instantiating (the now deprecated) `Symfony\Component\Cache\Simple\FilesystemCache`.

**Comment by AndrzejDunowski** [comment]
My application used Simple Cache, which is now deprecated. 
I corrected it, according to Cache Contracts. 

I think that while Simple Cache exists, `psr/simple-cache` should exist in dependencies.

**Comment by derrabus** [comment]
> I think that while Simple Cache exists, `psr/simple-cache` should exist in dependencies.

Either that or we need to make sure that the deprecated classes remain usable without the PSR interface.

**Comment by nicolas-grekas** [comment]
Oh, it's already in the changelog and upgrade files, all good on my side, there is no other way I know about.

**Comment by nicolas-grekas** [comment]
> make sure that the deprecated classes remain usable without the PSR interface

that's not possible, as the very purpose of the classes in the `Simple` namespace is to pass the psr/simple-cache type hints.

**Comment by derrabus** [comment]
> the very purpose of the classes in the `Simple` namespace is to pass the psr/simple-cache type hints.

Fair enough. In the project where I fixed this issue a couple of days ago, the developers just used `FilesystemCache` standalone because it's a cache that did what they needed it to do. They didn't really care about the fact that it implemented the PSR interface.

I might have missed the discussion on the topic, but would it be that problematic to keep the dependency for now and remove it together with the deprecated implementations in 5.0? I get the impression that the upgrade path might be smoother that way since developers would get a proper deprecation notice instead of a fatal error.

**Comment by nicolas-grekas** [comment]
That'd smoothen for ppl that move to 4.3 and do the change before 4.4. Thus it wouldn't change the core process. But we can yes.

**Comment by nicolas-grekas** [comment]
Closing as "not a bug". We can smoothen, but the more we wait the less an issue this is :)
This also matches our policy regarding dependencies.

**Comment by Jean85** [comment]
@nicolas-grekas isn't that a breaking change though? We are upgrading from the previous minor, and our apps are breaking due to this change...

**Comment by teohhanhui** [comment]
Mysterious failure on AppVeyor: https://ci.appveyor.com/project/dunglas/dunglasapibundle/builds/25778904#L340

Something different about Windows, I suppose.

**Comment by derrabus** [comment]
@teohhanhui Could be a variant of #32395

**Comment by teohhanhui** [comment]
Curiously enough, it doesn't happen on `master` branch of API Platform, but only `2.4`. Must be due to a different combination of package versions...

**Comment by tacman** [comment]
TL;DR:   composer req psr/simple-cache
